inflight.go

```diff
@@ -89,15 +83,26 @@ func (c *Client[T]) endBatchFlight(ids []string, keyFn KeyFn, call *inFlightCall
 	c.inFlightBatchMutex.Unlock()
 }
 
-type makeBatchCallOpts[T, V any] struct {
+type makeBatchCallOpts[T any] struct {
 	ids   []string
-	fn    BatchFetchFn[V]
+	fn    BatchFetchFn[T]
 	keyFn KeyFn
 	call  *inFlightCall[map[string]T]
 }
 
-func makeBatchCall[T, V any](ctx context.Context, c *Client[T], opts makeBatchCallOpts[T, V]) {
+func makeBatchCall[T any](ctx context.Context, c *Client[T], opts makeBatchCallOpts[T]) {
 	response, err := opts.fn(ctx, opts.ids)
+	for id, record := range response {
+		// We never want to discard values from the fetch functions, even if they
+		// return an error. Instead, we'll pass them to the user along with any
+		// errors and let them decide what to do.
+		opts.call.val[id] = record
+		// However, we'll only write them to the cache if the fetchFunction returned a non-nil error.
```

Comment:
Is there a typo in this comment? We only write them to the cache if the fetchFunction *didn't* return a non-nil error.

(except for errOnlyDistributedRecords)